src/main/kotlin/com/example/daitssuapi/domain/course/service/CourseService.kt

```diff
@@ -172,4 +163,33 @@ class CourseService(
                 updatedAt = it.course.updatedAt
             )
         }
+    
+    fun updateCalendar(calendarRequest: CalendarRequest, calendarId : Long) : CalendarResponse {
+        val calendar = calendarRepository.findByIdOrNull(calendarId)
+            ?: throw DefaultException(ErrorCode.CALENDAR_NOT_FOUND)
+        
+        val dateTime = checkDate(calendarRequest.dueAt)
+        
+        calendar.updateCalendar(calendarRequest = calendarRequest, dueAt = dateTime)
+            .also { calendarRepository.save(calendar) }
+        
+        return CalendarResponse(
+            id = calendar.id,
+            type = calendar.type,
+            dueAt = calendar.dueAt,
+            name = calendar.name
+        )
```

Comment:
```suggestion
return calendar
            .also { 
                it.name = calendarRequest.name
                it.type = calendarRequest.type
                it.course = calendarRequest.course
                it.dueAt = dateTime
            }
            .let {
                CalendarResponse(
                    id = calendar.id,
                    type = calendar.type,
                    dueAt = calendar.dueAt,
                    name = calendar.name
                )
            }
```

요런 느낌으로 하면 저장 후 리턴이 안될까요?

Reply:
아마 안될꺼 같습니다 달아주신 also문이 updateCalendar 함수와 같은 실행인데
save를 안하면 조회시 데이터가 반영이 안되어있더라구요!